vault/resource_mongodbatlas_secret_role.go

```diff
@@ -0,0 +1,238 @@
+				Type:        schema.TypeString,
+				Required:    false,
+				Description: "ID for the project to which the target API Key belongs",
+			},
+			"roles": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "List of roles that the API Key needs to have",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"id_addresses": {
+				Type:        schema.TypeList,
+				Required:    false,
+				Description: "IP address to be added to the whitelist for the API key",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"cidr_blocks": {
+				Type:        schema.TypeList,
+				Required:    false,
+				Description: "Whitelist entry in CIDR notation to be added for the API key",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"project_roles": {
+				Type:        schema.TypeList,
+				Required:    false,
+				Description: "Roles assigned when an org API key is assigned to a project API key",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"ttl": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Duration in seconds after which the issued credential should expire",
+			},
+			"max_ttl": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The maximum allowed lifetime of credentials issued using this role",
+			},
+		},
+	}
+}
+
+func mongodbAtlasSecretRoleCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	backend := d.Get(consts.FieldBackend).(string)
+	name := d.Get("name").(string)
+
+	path := backend + "/roles/" + name
+	log.Printf("[DEBUG] Creating role %q in MongoDB Atlas", name)
+
+	data := map[string]interface{}{}
+	if v, ok := d.GetOk("organization_id"); ok {
+		data["organization_id"] = v
+	}
+	if v, ok := d.GetOk("project_id"); ok {
+		data["project_id"] = v
+	}
+	if v, ok := d.GetOk("roles"); ok {
+		data["roles"] = v
+	}
+	if v, ok := d.GetOk("ip_addresses"); ok {
+		data["ip_addresses"] = v
+	}
+	if v, ok := d.GetOk("cidr_blocks"); ok {
+		data["cidr_blocks"] = v
+	}
+	if v, ok := d.GetOk("project_roles"); ok {
+		data["project_roles"] = v
+	}
+	if v, ok := d.GetOk("ttl"); ok {
+		data["ttl"] = v
+	}
+	if v, ok := d.GetOk("max_ttl"); ok {
+		data["max_ttl"] = v
+	}
+
+	if _, err := client.Logical().Write(path, data); err != nil {
+		return diag.Errorf("error updating MongoDB Atlas role %q, err=%s", name, err)
+	}
+
+	d.SetId(path)
+
+	return mongodbAtlasSecretRoleRead(ctx, d, meta)
+}
+
+func mongodbAtlasSecretRoleRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	path := d.Id()
+	if err := d.Set(consts.FieldPath, path); err != nil {
```

Comment:
We will also need to set the `backend` field to state here. We can probably do that by extracting the value using a regex. An example of constructing such a regex can be found [here](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_azure_auth_backend_role.go#L20), and the usage of this regex to obtain the backend can be found [here](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_azure_auth_backend_role.go#L336-L345) 😄

Reply:
Got it! Thank you for the examples, very helpful! 🙏